src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -261,6 +261,16 @@ type FSharpSymbolUse(denv: DisplayEnv, symbol: FSharpSymbol, inst: TyparInstanti
     member this.IsPrivateToFile =
         let isPrivate =
             match this.Symbol with
+            | :? FSharpMemberOrFunctionOrValue as m when not m.IsModuleValueOrMember ->
+
+                // In case it's a parameter and there is a signature file, then it's not private
+
+                // TODO: Can it be anything else than a parameter?
+
+                // TODO: Is there a way to tell there is a signature file?
+
+                // Since we don't know any better, we have to assume it's not private
```

Comment:
Maybe we could record additional info about the value, like it was done in https://github.com/dotnet/fsharp/pull/10327?

Reply:
I'll check that out, thanks. That could work. 

Another option I was thinking about is to fix the Symbol's `SignatureLocation` which is currently always pointing to the implementation file. (Not sure how yet.)